### PR TITLE
US 52

### PR DESCRIPTION
**Pull request by BrennanAyers**
Most of this was Billy and Alex.
This is essentially the full implementation of Strava Webhooks, we will just need to change the outputs and math onto UserSongs PR. For proof of concept, it's random.

### CODE REVIEW

File: app/controllers/hooks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
+class HooksController < ApplicationController
+  skip_before_action :verify_authenticity_token
+
+  def strava
+    activity = JSON.parse(request.body.read, symbolize_names: true)
+
+    if activity[:aspect_type] == 'create'
```
[the remark this answers is not in the document]
Comment (BrennanAyers) [reply app/controllers/hooks_controller.rb]: Next time buddy.

File: app/controllers/hooks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      SongSifter.new(songs, user).sift_songs
+    end
+  end
+
+  def stream_hash(response)
+    smashed_hash = {}
+    i = 0
+    response[3][:data].map do |time|
+      smashed_hash[time] = {}
+      smashed_hash[time][response[0][:type]] = response[0][:data][i]
```
Comment (alexander-mathieu) [inline app/controllers/hooks_controller.rb]: That is one gnarly hash smasher.

File: app/services/strava_service.rb [hunk 3]
```diff
@@ -13,6 +13,10 @@ def get_user_activities
     get_json('activities')
   end
 
+  def get_user_activity(activity_id)
+    get_json("activities/#{activity_id}")
```
Comment (alexander-mathieu) [inline app/services/strava_service.rb]: 👍

File: spec/fixtures/strava_webhook_create_activity.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
+  "aspect_type": "create",
+  "event_time": 1564011114,
+  "object_id": 2560643329,
+  "object_type": "activity",
+  "owner_id": 44312407,
```
Comment (alexander-mathieu) [inline spec/fixtures/strava_webhook_create_activity.json]: I'm glad fixture fever is catching on.

File: config/routes.rb [hunk 5]
```diff
@@ -12,6 +12,7 @@
   get 'spotify/songs', to: 'spotify#index'
   post '/playlist', to: 'playlist#create'
 
+  post 'hooks/strava', to: 'hooks#strava'
```
Comment (alexander-mathieu) [inline config/routes.rb]: Webhooks!

File: app/controllers/hooks_controller.rb [hunk 6]
```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+# Top Level Documentation
+class HooksController < ApplicationController
+  skip_before_action :verify_authenticity_token
```
Comment (alexander-mathieu) [inline app/controllers/hooks_controller.rb]: I'm curious if there's another way around this.

**Pull request merged**